XamlControlsGallery/ControlExample.xaml.cs

```diff
@@ -350,5 +366,177 @@ private void OnValueChanged(ControlExampleSubstitution sender, object e)
         {
             GenerateAllSyntaxHighlightedContent();
         }
+
+        private void ScreenshotButton_Click(object sender, RoutedEventArgs e)
+        {
+            TakeScreenshot();
+        }
+
+        private void ScreenshotDelayButton_Click(object sender, RoutedEventArgs e)
+        {
+            TakeScreenshotWithDelay();
+        }
+
+        private async void TakeScreenshot()
+        {
+            // Using RTB doesn't capture popups; but in the non-delay case, that probably isn't necessary.
+            // This method seems more robust than using AppRecordingManager and also will work on non-desktop devices.
+
+            RenderTargetBitmap rtb = new RenderTargetBitmap();
+            await rtb.RenderAsync(ControlPresenter);
+
+            var pixelBuffer = await rtb.GetPixelsAsync();
+            var pixels = pixelBuffer.ToArray();
+
+            var file = await UIHelper.ScreenshotStorageFolder.CreateFileAsync(GetBestScreenshotName(), CreationCollisionOption.ReplaceExisting);
+            using (var stream = await file.OpenAsync(FileAccessMode.ReadWrite))
+            {
+                var displayInformation = DisplayInformation.GetForCurrentView();
+                var encoder = await BitmapEncoder.CreateAsync(BitmapEncoder.PngEncoderId, stream);
+                encoder.SetPixelData(BitmapPixelFormat.Bgra8,
+                    BitmapAlphaMode.Premultiplied,
+                    (uint)rtb.PixelWidth,
+                    (uint)rtb.PixelHeight,
+                    displayInformation.RawDpiX,
+                    displayInformation.RawDpiY,
+                    pixels);
+
+                await encoder.FlushAsync();
+            }
+        }
+
+        public async void TakeScreenshotWithDelay()
+        {
+            // 3 second countdown
+            for (int i = 3; i > 0; i--)
+            {
+                ScreenshotStatusTextBlock.Text = i.ToString();
+                await Task.Delay(1000);
+            }
+            ScreenshotStatusTextBlock.Text = "Image captured";
+
+            // AppRecordingManager is desktop-only, and its use here is quite hacky,
+            // but it is able to capture popups (though not theme shadows).
+
+            bool isAppRecordingPresent = ApiInformation.IsTypePresent("Windows.Media.AppRecording.AppRecordingManager");
+            if (isAppRecordingPresent)

```

Comment:
If the AppRecordingManager is not present, maybe we should have a fallback in place, instead of just doing nothing. How about calling `TakeScreenshot` in that case?

Reply:
Good call, I did just that.